src/Extensibility.Kubernetes/KubernetesProvider.cs

```diff
@@ -19,8 +19,7 @@ namespace Extensibility.Kubernetes
 {
     public class KubernetesProvider : IExtensibilityProvider
     {
-        public async Task<DeleteResponse> Delete(DeleteRequest request, CancellationToken cancellationToken)
-        {
+        public async Task<DeleteResponse> Delete(DeleteRequest request, CancellationToken cancellationToken) => await DoWithUnhandledErrorHandling<DeleteResponse>(async () => {
```

Comment:
I don't think this needs to be an async method since the result of `DoWithUnhandledErrorHandling` is already an awaitable task and can be passed back to the caller directly.